### DISCUSSION TITLE
AdBlock plugin

### DISCUSSION THREAD

**Discussion by akmalkadi** [intersimone999/pyqtws#23] [Ideas]
Is it possible to support firefox or google chrome plugins? I would like to use AdBlock.

For example, block youtube ads using AdBlock

It will be really helpful to enable the browsers' extensions ( For example, AdBlock and Grammarly are commonly required)

**Comment by intersimone999** [comment]
Unfortunately, Silos does not support Firefox/Chrome plugins out of the box.

I think that the development of a plugin for Grammarly would be great and very helpful for many apps.

Instead, I think that including an ad blocker in Silos would be unethical in basically all the scenarios (even illegal, in some cases). 
The way I see it, using an ad blocker is totally fine in general-purpose web browsers, through which a user can reach _everything_, also websites with aggressive ads. In other words, the main use case scenario for which I think an ad blocker is ethical is for blocking unwanted ads from randomly visited websites.

Silos is not a general-purpose web browser: An ad blocker in Silos would target a specific service. For example, the YouTube silo allows to reach YouTube only and, therefore, an ad blocker for it would be used specifically for that service. In this specific example, YouTube offers a premium version that allows removing ads. An ad blocker for the YouTube silo would be a free YouTube premium (I think it would be illegal). For apps that do not provide a premium version without ads, I think it would be at least unethical: ads would be the only means to economically support the app. A website with aggressive ads would probably never become a silo in the first place; even if it did, users that don't like it because it contains too many ads could simply uninstall it.

**TL;DR**: I do not see any scenario in which an ad blocker for Silos would be ethical.

Edit: typos
